src/io/flutter/devtools/DevToolsManager.java

```diff
@@ -307,10 +347,36 @@ public void processTerminated(@NotNull ProcessEvent event) {
 
   public void openBrowserAndConnect(String serviceProtocolUri, String page) {
     BrowserLauncher.getInstance().browse(
-      DevToolsUtils.generateDevToolsUrl(devtoolsHost, devtoolsPort, serviceProtocolUri, page),
+      DevToolsUtils.generateDevToolsUrl(devtoolsHost, devtoolsPort, serviceProtocolUri, page, false, null),
       null
     );
   }
+
+  public void openPanel(String serviceProtocolUri, ContentManager contentManager, String tabName, String pageName) {
+    String url = DevToolsUtils.generateDevToolsUrl(devtoolsHost, devtoolsPort, serviceProtocolUri, null, true, pageName);
+
+    EngineOptions options =
+      EngineOptions.newBuilder(HARDWARE_ACCELERATED).build();
+    Engine engine = Engine.newInstance(options);
+    Browser browser = engine.newBrowser();
+
+    final Content content = contentManager.getFactory().createContent(null, tabName, false);
+
+    SwingUtilities.invokeLater(() -> {
+      // Creating Swing component for rendering web content
+      // loaded in the given Browser instance.
+      BrowserView view = BrowserView.newInstance(browser);
```

Comment:
because it could have been some time from when the action was requested until we are ready to launch, we need to make sure the contentManager has not been disposed or perhaps that the project is still valid.